wkcuber/utils.py

```diff
@@ -152,3 +152,84 @@ def time_stop(identifier):
 def wait_and_ensure_success(futures):
     for fut in concurrent.futures.as_completed(futures):
         fut.result()
+
+
+class BufferedSliceWriter(object):
+    def __init__(
+        self, dataset_path, layer_name, dtype, bounding_box, origin, buffer_size=32
+    ):
+
+        self.dataset_path = dataset_path
+        self.layer_name = layer_name
+        self.buffer_size = buffer_size
+
+        layer_path = path.join(self.dataset_path, self.layer_name, "1")
+
+        self.dataset = wkw.Dataset.open(layer_path, wkw.Header(dtype))
+        self.origin = origin
+        self.bounding_box = bounding_box
+
+        self.buffer = []
+        self.current_z = None
+        self.buffer_start_z = None
+
+    def write_slice(self, z: int, data: np.ndarray):
+        """Takes in a slice in [y, x] shape, writes to WKW file."""
+
+        if len(self.buffer) == 0:
+            self.current_z = z
+            self.buffer_start_z = z
+
+        assert (
+            z == self.current_z
+        ), "({}) Slices have to be written sequentially!".format(getpid())
```

Comment:
I don't think we need to print the pid in the logs

Reply:
Done :grin: